### PR TITLE
Update Essentials Battery API Docs

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change

Moves the external comments to inline and minor improvements.

### Issues Fixed

Related to #3960

### CODE REVIEW

File: src/Essentials/src/Battery/Battery.shared.cs [hunk 1]
```diff
@@ -132,89 +178,132 @@ void OnEnergySaverChanged(EnergySaverStatusChangedEventArgs e)
 			=> EnergySaverStatusChangedInternal?.Invoke(null, e);
 	}
 
-	/// <include file="../../docs/Microsoft.Maui.Essentials/BatteryState.xml" path="Type[@FullName='Microsoft.Maui.Essentials.BatteryState']/Docs/*" />
+	/// <summary>
+	/// Enumerates states of the battery.
```
Comment (hartez) [inline src/Essentials/src/Battery/Battery.shared.cs:182]:
```suggestion
	/// Describes possible states of the battery.
```

**Review by hartez — CHANGES_REQUESTED** [review]
Looks good, couple of suggestions.

**Review by hartez — APPROVED** [review]

**Review by jsuarezruiz — APPROVED** [review]

**Pull request merged**